src/resources/js/cytoscape/index.js

```diff
@@ -611,11 +516,103 @@ function drawGraph(){
             };
             highlightNextEle();
         }
+        function clearDijkstra() {
+            cy.elements().removeClass('highlighted');
+        }
+        function generateNodeStyle(){
+            let nodeStyle = [];
+            let displayed = getDisplayedValue(listNodeTypeId);
+            // 3. Generate the style
+            listNodeType.forEach(function(nodeType){
+                //generate a random color
+                let shape = nodeType.shape ?? 'diamond';
+                let value = nodeType[displayed].eng;
+                let color = undefined;
+                if(listHasChanged){
+                    color = nodeType.color ?? '#'+(Math.random()*0xFFFFFF<<0).toString(16);
+                }
+                else{
+                    color = nodeType.color ?? list_node_type_colors[value];
+                }
+                list_node_type_colors[value] = color;
+                let style = {
+                    selector: 'node[type = "'+value+'"]',
+                    style: {
+                        'background-color': color,
+                        'label': 'data(name)',
+                        'font-size': 25,
+                        'shape': shape
+                    }
+                };
+                nodeStyle.push(style);
+            });
+            return nodeStyle;
+        }
+        function updateStyle(){
+            // remove the old style
+            cy.style().resetToDefault();
+            // add the new style
+            let nodeStyle = generateNodeStyle();
 
-    });
+            // concat style and nodeStyle to get the new style
+            let newStyle = style.concat(nodeStyle);
 
+            cy.style(newStyle);
+        }
 
-}
+        // According to the list, we change the 'type' of each node.
+        // It uses the window.persons variable
+        function updateData(){
+            // foreach nodes
+            cy.nodes().forEach(function(node){
+               //update the data
+                let type = window.persons[node.id()][window.field_list[listNodeTypeId]];
```

Comment:
Récupérer plus d'informations au moment de récupérer la list ?

Reply:
fix in commit #6cfbd9c1c7afea4ad36d6b36884a9f52fb8adfec